File: src/0661.image-smoother.661/solution.ts
function imageSmoother (M: number[][]): number[][] {
    const R = M.length;
    const C = M[0].length;
    const result:number[][] = new Array(R);
    for (let i = 0; i < R; i++) {
        result[i] = new Array(C);
        for (let j = 0; j < C; j++) {
            let sum = 0;
            let count = 0;
            for (let dx = -1; dx < 2; dx++) {
                const x = i + dx;
                if (x === -1 || x === R) {
                    continue;
                }
                for (let dy = -1; dy < 2; dy++) {
                    const y = j + dy;
                    if (y === -1 || y === C) {
                        continue;
                    }
                    sum += M[x][y];
                    count++;
                }
            }
            result[i][j] = Math.floor(sum / count);
        }
    }

    return result;
}

File: src/0661.image-smoother.661/solution.java
class Solution {
    public int[][] imageSmoother(int[][] img) {
        int M = img.length;
        int N = img[0].length;
        int[][] result = new int[M][N];
        for(int i=0;i<M;i++){
            for(int j=0;j<N;j++){
                
                int count = 0;
                int sum = 0;
                for(int x=i-1;x<=i+1;x++){
                    for(int y=j-1;y<=j+1;y++){
                        if(x == -1 || y == -1 || x == M || y == N){
                            continue;
                        }
                        count++;
                        sum += img[x][y];
                    }
                }
                result[i][j] = sum/count;
            }
            
        }
        
        
        return result;
    }
}